types/shares.go

```diff
@@ -56,35 +56,87 @@ func (m Message) MarshalDelimited() ([]byte, error) {
 	return append(lenBuf[:n], m.Data...), nil
 }
 
-func appendToShares(shares []NamespacedShare, nid namespace.ID, rawData []byte, shareSize int) []NamespacedShare {
-	if len(rawData) < shareSize {
-		rawShare := rawData
-		paddedShare := zeroPadIfNecessary(rawShare, shareSize)
+// appendToSharesContiguous appends one raw data separately as shares
+// Used for messages
+func appendToShares(shares []NamespacedShare, nid namespace.ID, rawData []byte) []NamespacedShare {
+	const adjustedSize = ShareSize - NamespaceSize
+	if len(rawData) < adjustedSize {
+		rawShare := []byte(append(nid, rawData...))
+		paddedShare := zeroPadIfNecessary(rawShare, ShareSize)
+		share := NamespacedShare{paddedShare, nid}
+		shares = append(shares, share)
+	} else { // len(rawData) >= adjustedSize
+		shares = append(shares, split(rawData, nid)...)
+	}
+	return shares
+}
+
+// appendToSharesContiguous appends multiple raw data contiguously as shares
+// Used for transactions, intermediate state roots, and evidence
+func appendToSharesContiguous(shares []NamespacedShare, nid namespace.ID, rawDatas [][]byte) []NamespacedShare {
```

Comment:
I'm wondering if we shouldn't remove the first argument: `shares`? 

Everywhere this method is used it follows the pattern:
```go
shares := make([]NamespacedShare, 0)
// do stuff
// ... stuff
shares = appendToSharesContiguous(shares, IntermediateStateRootsNamespaceID, rawDatas) // or some other Namespace
return shares
```
Couldn't this be made into:
```
// do stuff
// ... stuff
shares := appendToSharesContiguous(IntermediateStateRootsNamespaceID, rawDatas)
return shares
```

Reply:
Not a bad idea

Reply:
Fixed in e0c0940.